driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -293,14 +292,19 @@ public Boolean isIdempotent() {
     @Override
     public String toString() {
         try {
-            if (forceNoValues) {
+            if (forceNoValues)
                 return getQueryString();
+            // 1) try first with all values inlined (will not work if some values require custom codecs,
+            // or if the required codecs are registered in a different CodecRegistry instance than the default one)
+            return maybeAddSemicolon(buildQueryString(null, CodecRegistry.DEFAULT_INSTANCE)).toString();
+        } catch (RuntimeException e1) {
+            // 2) try next with bind markers for all values to avoid usage of custom codecs
+            try {
+                return maybeAddSemicolon(buildQueryString(new ArrayList<Object>(), CodecRegistry.DEFAULT_INSTANCE)).toString();
+            } catch (RuntimeException e2) {
+                // Ugly but we have absolutely no context to get the registry from
+                return String.format("built query (could not generate with default codec registry: %s)", e2.getMessage());
```

Comment:
Is this final case even possible now if we don't try parsing the values?  I guess it could depend on the implementation of `BuiltStatement`, but none of the ones we define do from what I can tell.  Still seems like a good idea to have this here just in case.